src/main/java/macrobase/analysis/contextualoutlier/Context.java

```diff
@@ -1,185 +1,245 @@
 package macrobase.analysis.contextualoutlier;
 
 import java.util.ArrayList;
+import java.util.BitSet;
+import java.util.HashSet;
 import java.util.List;
-import java.util.SortedMap;
-import java.util.TreeMap;
+import java.util.Map;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import macrobase.analysis.stats.BatchTrainScore;
 import macrobase.datamodel.Datum;
 import macrobase.ingest.DatumEncoder;
 
 
 public class Context {
     private static final Logger log = LoggerFactory.getLogger(Context.class);
 
-    
+
+    /**
+     * A list of ordered contextual dimensions this node represents
+     * and intervals for each dimension
+     */
+    private List<Integer> dimensions = new ArrayList<Integer>();
+    private List<Interval> intervals = new ArrayList<Interval>();
+    private int size = -1;
+    private List<Context> parents = new ArrayList<Context>();
+    private HashSet<Context> oneDimensionalAncestors = new HashSet<Context>();
+    //Sample datums in this context
+    private HashSet<Datum> sample = new HashSet<Datum>();
+    //the outlier detector used for detection in this context
+    private BatchTrainScore detector;
+
+
+    /**
+     * Global context
+     */
+    public Context(HashSet<Datum> sample) {
+        this.sample = sample;
+    }
+
+    /**
+     * Initialize a one dimensional context
+     *
+     * @param dimension
+     * @param interval
+     * @param parent
+     */
+    public Context(int dimension, Interval interval, Context parent) {
+        dimensions.add(dimension);
+        intervals.add(interval);
+        parents.add(parent);
+
+        for (Datum d : parent.sample) {
+            if (containDatum(d)) {
+                sample.add(d);
+            }
+        }
+
+        oneDimensionalAncestors.add(this);
+
```

Comment:
lots of empty/double spaces. before you merge, can you run intellij's reformat code on these files?